join.c

```diff
@@ -117,20 +118,27 @@ void joinFreeCallback(void *privdata)
  */
 void joinIdleCallback(Connection *conn)
 {
+    RedisRaftCtx *rr = ConnGetRedisRaftCtx(conn);
     JoinState *state = ConnGetPrivateData(conn);
 
+    time_t now;
+    time(&now);
+
+    // This value should probably be dynamic and be considered a multiple of election time period? 2x?
+    // need to figure that out (i.e. perhaps     rr->config->election_timeout*2 ?
+    if (difftime(now, state->start) > 10) {
+        LOG_ERROR("timed out trying to connect");
```

Comment:
Consider something more detailed like "Cluster join: timed out trying to connect to <host>:<port>"

Reply:
this isn't an individual connection timeout, so the only option would be to try to stringify the whole node addr list, I'm not sure that's right  put a message that it time out as too longer than the "join_timeout" # of seconds